src/ray/raylet/local_lease_manager.cc

```diff
@@ -1000,26 +1000,28 @@ void LocalLeaseManager::Grant(
   } else {
     allocated_resources = worker->GetAllocatedInstances();
   }
-  ::ray::rpc::ResourceMapEntry *resource;
-  for (auto &resource_id : allocated_resources->ResourceIds()) {
-    bool first = true;  // Set resource name only if at least one of its
-                        // instances has available capacity.
-    auto instances = allocated_resources->Get(resource_id);
-    for (size_t inst_idx = 0; inst_idx < instances.size(); inst_idx++) {
-      if (instances[inst_idx] > 0.) {
-        if (first) {
-          resource = reply->add_resource_mapping();
-          resource->set_name(resource_id.Binary());
-          first = false;
+  for (const auto &reply_callback : reply_callbacks) {
+    ::ray::rpc::ResourceMapEntry *resource;
+    for (auto &resource_id : allocated_resources->ResourceIds()) {
```

Comment:
Might be missing something, but it looks like these loops should be inverted -- nothing about the inner loop logic depends on which callback we are iterating through. So you can make a single pass through `allocated_resources->ResourceIds()` and populate all callbacks' resource mappings at once instead.